x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -1922,6 +1922,115 @@ public void testPruneRenameOnAggBy() {
         var source = as(agg.child(), EsRelation.class);
     }
 
+    /**
+     * Expects
+     * Project[[c1{r}#2, c2{r}#4, cs{r}#6, cm{r}#8, cexp{r}#10]]
+     * \_Eval[[c1{r}#2 AS c2, c1{r}#2 AS cs, c1{r}#2 AS cm, c1{r}#2 AS cexp]]
+     *   \_Limit[500[INTEGER]]
+     *     \_Aggregate[[],[COUNT([2a][KEYWORD]) AS c1]]
+     *       \_EsRelation[test][_meta_field{f}#17, emp_no{f}#11, first_name{f}#12, ..]
+     */
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/100634")
+    public void testEliminateDuplicateAggsCountAll() {
+        var plan = plan("""
+              from test
+            | stats c1 = count(1), c2 = count(2), cs = count(*), cm = count(), cexp = count("123")
+            """);
+
+        var project = as(plan, Project.class);
+        assertThat(Expressions.names(project.projections()), contains("c1", "c2", "cs", "cm", "cexp"));
+        var eval = as(project.child(), Eval.class);
+        var fields = eval.fields();
+        assertThat(Expressions.names(fields), contains("c2", "cs", "cm", "cexp"));
+        for (Alias field : fields) {
+            assertThat(Expressions.name(field.child()), is("c1"));
+        }
+        var limit = as(eval.child(), Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        var aggs = agg.aggregates();
+        assertThat(Expressions.names(aggs), contains("c1"));
```

Comment:
> Shouldn't we also assert that there is only 1 aggregation?
This is exactly what this and the following line assert - there's only one aggregation c1, which is an alias for Count.

Reply:
We check that there is just one aggregation node, but the point of this PR is that the aggregation node contains just 1 aggregation expression, no?

To be more precise, I think the test is missing
```java
        assertEquals(1, aggs.size());
```
otherwise, we only assert that `"c1"` is contained, not that it is the only aggregation expression.

Reply:
contains is a strict checker - it checks the content, order and length.
>  For a positive match, the examined iterable must be of the same length as the number of specified items

Reply:
D'oh, thanks for clarifying that!